drivers/counter/timer_dtmr_cmsdk_apb.c

```diff
@@ -87,26 +88,61 @@ static u32_t timer_dtmr_cmsdk_apb_read(struct device *dev)
 	return value;
 }
 
+static int timer_dtmr_cmsdk_apb_set_wrap_alarm(struct device *dev,
+					       u32_t ticks,
+					       counter_wrap_callback_t callback,
+					       void *user_data)
+{
+	const struct timer_dtmr_cmsdk_apb_cfg * const cfg =
+						dev->config->config_info;
+	struct timer_dtmr_cmsdk_apb_dev_data *data = dev->driver_data;
+
+	/* Set callback */
+	data->user_cb = callback;
+
+	/* Store the reload value */
+	data->load = ticks;
+
+	/* Set the timer to count */
+	cfg->dtimer->timer1load = ticks;
+
+	/* Enable IRQ */
+	cfg->dtimer->timer1ctrl |= (DUALTIMER_CTRL_INTEN
+				    | DUALTIMER_CTRL_MODE);
+
+	return 0;
+}
+
+static u32_t timer_dtmr_cmsdk_apb_get_wrap(struct device *dev)
+{
+	struct timer_dtmr_cmsdk_apb_dev_data *data = dev->driver_data;
+
+	u32_t ticks = data->load;
+
+	return ticks;
+}
+
 static int timer_dtmr_cmsdk_apb_set_alarm(struct device *dev,
-					   counter_callback_t callback,
-					   u32_t count, void *user_data)
+				const struct counter_alarm_cfg *alarm_cfg)
 {
 	const struct timer_dtmr_cmsdk_apb_cfg * const cfg =
 						dev->config->config_info;
 	struct timer_dtmr_cmsdk_apb_dev_data *data = dev->driver_data;
 
-	if (data->status == DTIMER_DISABLED) {
-		return -ENOTSUP;
-	}
+	if (alarm_cfg->handler == NULL)
+		return -ENODEV;
 
 	/* Set callback */
-	user_cb = callback;
+	data->user_handler = alarm_cfg->handler;
+
+	/* Set Alarm Cfg */
+	data->a_cfg = alarm_cfg;
 
 	/* Store the reload value */
-	data->load = count;
+	data->load = alarm_cfg->ticks;
```

Comment:
that looks suspicious. load value looks like wrap (and it is modified in `set_wrap` function). `set_alarm` should configure compare channels if timer peripheral has any. Note that `set_wrap` is equivalent of legacy `set_alarm` function and `set_channel_alarm` is something new.